git format-patch --stdout HEAD^..HEAD | sed -e 's| a/| libreoffice-3.4.1/libreoffice-build/|' -e 's| b/| libreoffice-3.4.1/libreoffice-build/|' > ~/libreoffice-build-161a3f9f2229675e083e0806e8171cb0b3145168.diff
From 161a3f9f2229675e083e0806e8171cb0b3145168 Mon Sep 17 00:00:00 2001
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Fri, 1 Jul 2011 17:16:00 +0200
Subject: [PATCH] add generic Ubuntu distro

---
 distro-configs/Ubuntu.conf.in |   17 +++++++++++++++++
 patches/dev300/apply          |    1 +
 2 files changed, 18 insertions(+), 0 deletions(-)
 create mode 100644 distro-configs/Ubuntu.conf.in

Index: libreoffice-3.4.1/libreoffice-build/patches/dev300/apply
===================================================================
--- libreoffice-3.4.1.orig/libreoffice-build/patches/dev300/apply	2011-07-15 11:43:05.776851060 +0200
+++ libreoffice-3.4.1/libreoffice-build/patches/dev300/apply	2011-07-15 11:43:09.096851091 +0200
@@ -54,14 +54,9 @@
 Debian : DebianBase, DebianOnly, DebianSidOnly, Debian32Only
 DebianEtch : DebianBase, DebianOnly, DebianEtchOnly, Debian32Only
 # Ubuntu
-UbuntuBased : DebianBase, UbuntuOnly
-UbuntuHardy : UbuntuBased, UbuntuHardyOnly
-UbuntuJaunty : UbuntuBased, UbuntuJauntyOnly
-UbuntuKarmic : UbuntuBased, UbuntuKarmicOnly
-UbuntuLucid : UbuntuBased, UbuntuLucidOnly
-UbuntuMaverick : UbuntuBased, UbuntuMaverickOnly
-UbuntuNatty : UbuntuBased, UbuntuNattyOnly
-UbuntuL10n : UbuntuBased, UbuntuL10nOnly
+UbuntuBased : DebianBase
+UbuntuL10n : UbuntuBased
+Ubuntu : UbuntuBased
 # DroplineGNOME
 DroplineGNOME : LinuxCommon, Mono, msaccess
 DroplineGNOME64 : LinuxCommon, Mono, msaccess
@@ -1016,23 +1011,6 @@
 # use -llpsolve55_pic
 system-lpsolve-use-_pic.diff, rengelha
 
-[ UbuntuOnly ]
-SectionOwner => ccheney
-
-ubuntu-palette.diff
-ubuntu-sparc-hack.diff, mklose
-ubuntu-no-stack-protector.diff, doko
-#gccXXXXX.solenv.javaregistration.diff
-ubuntu-mstopdf.diff, ccheney
-# FIXME: update for LO
-#f11-fullscreen.diff, ccheney
-
-[ UbuntuOnly ]
-unxlngi6-notune.diff, doko
-
-[ UbuntuL10nOnly ]
-human-icons-i18n.diff, mklose
-
 [ DebianBaseOnly ]
 soffice-opengl-and-locking-config.diff, mklose
 
@@ -1188,40 +1166,6 @@
 [ Fixes ]
 external-apm-header.diff, freuter
 
-[ UbuntuHardyOnly ]
-# Add patch to only show local files needed when gnome-vfs/gio is disabled
-# FIXME: update for LO
-#ubuntu-gnome-fpicker-local-only.diff
-
-[ UbuntuJauntyOnly ]
-# Add patch to force use of gvfs fuse path needed when gnome-vfs/gio is disabled
-# FIXME: update for LO
-#ubuntu-gnome-fpicker-gfile-fuse.diff, ccheney
-
-[ UbuntuKarmicOnly ]
-# Add patch to force use of gvfs fuse path needed when gnome-vfs/gio is disabled
-# FIXME: update for LO
-#ubuntu-gnome-fpicker-gfile-fuse.diff, ccheney
-
-[ UbuntuLucidOnly ]
-# Add patch to force use of gvfs fuse path needed when gnome-vfs/gio is disabled
-# FIXME: update for LO
-#ubuntu-gnome-fpicker-gfile-fuse.diff, ccheney
-ubuntu-arm-thumb.diff, doko
-
-[ UbuntuMaverickOnly ]
-# Add patch to force use of gvfs fuse path needed when gnome-vfs/gio is disabled
-# FIXME: update for LO
-#ubuntu-gnome-fpicker-gfile-fuse.diff, ccheney
-ubuntu-arm-thumb.diff, doko
-
-[ UbuntuNattyOnly ]
-# Add patch to force use of gvfs fuse path needed when gnome-vfs/gio is disabled
-# FIXME: update for LO
-#ubuntu-gnome-fpicker-gfile-fuse.diff, ccheney
-# FIXME: still needed for GCC-4.5?
-#ubuntu-arm-thumb.diff, doko
-
 [ Experimental ]
 experimental_ooapi.diff, flr
 
@@ -1255,9 +1199,6 @@
 [ DebianSidOnly ]
 # link dynamically with liblpsolve55.so
 system-lpsolve-rpath.diff, rengelha
-[ UbuntuOnly ]
-# link dynamically with liblpsolve55.so
-system-lpsolve-rpath.diff, rengelha
 
 [ DebianBaseOnly ]
 # link with -lcolamd, needed because our liblpsolve55{,_pic}.a doesn't include it
